packages/tests/tests/waku.node.spec.ts

```diff
@@ -186,3 +186,38 @@ describe("Decryption Keys", () => {
     expect(receivedMsg.timestamp?.valueOf()).to.eq(messageTimestamp.valueOf());
   });
 });
+
+describe("User Agent", () => {
+  let waku: Waku;
+  let nwaku: Nwaku;
+
+  afterEach(async function () {
+    !!nwaku && nwaku.stop();
+    !!waku && waku.stop().catch((e) => console.log("Waku failed to stop", e));
+  });
+
+  it("Sets default value correctly", async function () {
+    this.timeout(20_000);
+    nwaku = new Nwaku(makeLogFileName(this));
+    await nwaku.start({
+      filter: true,
+      store: true,
+      lightpush: true,
+      persistMessages: true,
+    });
+    const multiAddrWithId = await nwaku.getMultiaddrWithId();
+
+    waku = await createLightNode({
+      staticNoiseKey: NOISE_KEY_1,
+      userAgent: "test-user-agent",
+    });
+    await waku.start();
+    await waku.dial(multiAddrWithId);
+    await waitForRemotePeer(waku);
+
+    const peerInfo = await waku.libp2p.peerStore.metadataBook.get(
+      waku.libp2p.peerId
```

Comment:
I am pretty sure the peer store only store info about other peers.
You are trying to get the info your your own node out of the peer store. I am surprised you are getting anything out of it.

A more appropriate test would be to connect 2 js-waku nodes together, set the user agent on one node and check from the **other node** whether it worked.